subscription-fee/src/subtract_payments.rs

```diff
@@ -0,0 +1,164 @@
+use core::hint::unreachable_unchecked;
+
+use auto_farm::common::{address_to_id_mapper::AddressId, unique_payments::UniquePayments};
+
+use crate::service::ServiceInfo;
+
+multiversx_sc::imports!();
+multiversx_sc::derive_imports!();
+
+#[must_use]
+#[derive(Debug, PartialEq, Eq, Clone, TopEncode, TopDecode, TypeAbi)]
+pub enum MyVeryOwnScResult<
+    T: NestedEncode + NestedDecode + TypeAbi,
+    E: NestedEncode + NestedDecode + TypeAbi,
+> {
+    Ok(T),
+    Err(E),
+}
+
+impl<T: NestedEncode + NestedDecode + TypeAbi, E: NestedEncode + NestedDecode + TypeAbi>
+    MyVeryOwnScResult<T, E>
+{
+    pub fn is_err(&self) -> bool {
+        matches!(*self, MyVeryOwnScResult::Err(_))
+    }
+
+    /// # Safety
+    ///
+    /// Calling this method on an [`Err`] is *[undefined behavior]*.
+    pub unsafe fn unwrap_unchecked(self) -> T {
+        match self {
+            MyVeryOwnScResult::Ok(t) => t,
+            MyVeryOwnScResult::Err(_) => unreachable_unchecked(),
+        }
+    }
+}
+
+#[multiversx_sc::module]
+pub trait SubtractPaymentsModule:
+    crate::fees::FeesModule
+    + crate::service::ServiceModule
+    + crate::pair_actions::PairActionsModule
+    + energy_query::EnergyQueryModule
+    + multiversx_sc_modules::ongoing_operation::OngoingOperationModule
+{
+    #[endpoint(subtractPayment)]
+    fn subtract_payment(
+        &self,
+        service_index: usize,
```

Comment:
so the caller can have multiple serviceIDs. Check if user is subscriber for that service ID and check the periodicity. Like when was the last time payment was subsctracted.